cbor2/encoder.py

```diff
@@ -114,6 +114,31 @@ def encode_map(encoder, value):
         encoder.encode(val)
 
 
+def encode_sortable_key(val):
+    '''Takes a key and calculates the length of it's optimal byte representation'''
+    if val.__class__ == bytes:
+        return len(val), val
+    if val.__class__ == unicode:
+        x = val.encode('utf-8')
+        return len(x), x
+    if val.__class__ == int:
```

Comment:
I would prefer `if type(val) is int:`. The `__class__` attribute may not even exist on some obscure built-in types. 

This of course applies to the previous lines too.

Reply:
I just copied the style used for dispatching on object types in CBOREncoder https://github.com/agronholm/cbor2/blob/67beccf3c738948d30cc76b7e37e0a350b2185a9/cbor2/encoder.py#L315